lua/entities/terminator_nextbot_zambiemechaelite/shared.lua

```diff
@@ -0,0 +1,422 @@
+
+ENT.WalkSpeed = 85
+ENT.MoveSpeed = 240
+ENT.RunSpeed = 420
+ENT.AccelerationSpeed = 1200
+
+ENT.FistDamageMul = 2.0
+ENT.MyPhysicsMass = 300
+ENT.JumpHeight = 300
+
+ENT.DuelEnemyDist = 500
+ENT.CloseEnemyDistance = 600
+
+ENT.HeightToStartTakingDamage = 500
+ENT.FallDamagePerHeight = 0.025
+ENT.DeathDropHeight = 3000
+
+ENT.term_SoundPitchShift = -45
+ENT.term_SoundLevelShift = 20
+
+ENT.Mecha_ShockwaveCooldown = 4
+ENT.Mecha_ShockwaveThreshold = 250
+ENT.Mecha_CriticalHealth = 0.25
+
+ENT.TERM_MODELSCALE = 2.25
+ENT.CollisionBounds = { Vector( -18, -18, 0 ), Vector( 18, 18, 90 ) }
+
+if CLIENT then
+    language.Add( "terminator_nextbot_zambiemechaelite", ENT.PrintName )
+
+    function ENT:Draw()
+        render.SetColorModulation( 0.4, 0.4, 0.6 )
+        self:DrawModel()
+        render.SetColorModulation( 1, 1, 1 )
+    end
+    
+    return
+end
+
+function ENT:AdditionalInitialize()
+    BaseClass.AdditionalInitialize( self )
+    
+    self:SetSubMaterial( 0, "phoenix_storms/cube" )
+    self:SetColor( Color( 35, 35, 55 ) )
+    
+    self.HasBrains = true
+    self.IsStupid = false
+    
+    self.Mecha_IsEnraged = false
+    self.Mecha_NextHealthRegen = 0
+end
+
+ENT.MyClassTask = {
+    DisableBehaviour = function( self, data )
+        local clockTime = CurTime() - terminator_Extras.Mecha_GlobalClock
+        local cycleTime = self.Mecha_MarchInterval + self.Mecha_StopInterval
+        local cycleProgress = clockTime % cycleTime
+        
+        if cycleProgress > self.Mecha_MarchInterval then
+            return true
+        end
+        
+        return false
+    end,
+    
+    BehaveUpdate = function( self, data )
+        local cur = CurTime()
+        
+        if self.Mecha_NextHealthRegen < cur then
+            self.Mecha_NextHealthRegen = cur + self.HealthRegenInterval
+            local newHealth = math.min( self:Health() + self.HealthRegen, self:GetMaxHealth() )
+            self:SetHealth( newHealth )
+        end
+        
+        if not self.Mecha_IsEnraged then
+            local healthPercent = self:Health() / self:GetMaxHealth()
+            if healthPercent <= self.Mecha_CriticalHealth then
+                self:EnterEnragedState()
+            end
+        end
+    end,
+    
+    OnLandOnGround = function( self, data, landedOn, height )
+        if height > self.Mecha_ShockwaveThreshold then
+            self:CreateEliteShockwave( height )
+        end
+    end,
+    
+    OnKilled = function( self, data, damage, rag )
+        self:EliteSelfDestruct()
+    end,
+    
+    OnDamaged = function( self, data, damage )
+        if damage:IsFallDamage() and damage:GetDamage() < 150 then
+            return true
+        end
+        
+        if not damage:IsFallDamage() and damage:GetDamage() > 20 and math.random( 1, 100 ) < 15 then
+            timer.Simple( 0.1, function()
+                if not IsValid( self ) then return end
+                self:CreateMiniShockwave()
+            end )
+        end

```

Comment:
You're creating shockwaves inside OnLandOnGround and OnDamaged?
Just handle shockwaves in OnLandOnGround, that way you won't ever double them up, and all the logic happens in one place